### ISSUE TITLE
load of misaligned address 0x000100000002 for type 'uint_fast8_t'

### ISSUE THREAD

**Issue by asarubbo** [mdadams/jasper#87]
On 1.900.27:

```
warning: trailing garbage in marker segment (14 bytes)
warning: trailing garbage in marker segment (2 bytes)
warning: trailing garbage in marker segment (14 bytes)
warning: trailing garbage in marker segment (14 bytes)
warning: trailing garbage in marker segment (14 bytes)
warning: trailing garbage in marker segment (14 bytes)
/tmp/portage/media-libs/jasper-1.900.27/work/jasper-1.900.27/src/libjasper/jpc/jpc_dec.c:819:43: runtime error: load of misaligned address 0x000100000002 for type 'uint_fast8_t' (aka 'unsigned char'), which requires 8 byte alignment
0x000100000002: note: pointer points here
<memory cannot be printed>
```

Testcase: https://github.com/asarubbo/poc/blob/master/00052-jasper-misalignedaddress

**Comment by mdadams** [comment]
The problem is also present in the most recent version of JasPer in my working tree (which is 1.900.29 with some minor changes).  ASAN reports a stack buffer overflow.  At least I can reproduce a failure on my machine, which is good news.  I'll have to investigate further to find the cause.

**Comment by asarubbo** [comment]
ok, I verified it. WIth asan is a stack overflow and is exactly the report in #85

**Comment by mdadams** [comment]
Since you have indicated that this is a duplicate of #85, I am marking this issue (i.e., #87) closed.

**Comment by mdadams** [comment]
Fixed as of commit 695cb182a25af946ca1102817b7a2b7af1d81a05.
